graylog2-web-interface/src/views/components/contexts/SearchPageLayoutProvider.tsx

```diff
@@ -14,46 +14,36 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+
 import * as React from 'react';
-import { useMemo } from 'react';
 
-import SearchPageLayoutContext from './SearchPageLayoutContext';
-import SearchPageLayoutState from './SearchPageLayoutState';
+import type { LayoutState } from 'views/components/contexts/SearchPageLayoutContext';
+import SearchPageLayoutContext, { FULL_MENU } from 'views/components/contexts/SearchPageLayoutContext';
 
-type Props = {
+type SearchPageLayoutProviderProps = {
   children: React.ReactNode
-  setLayoutState: (stateKey: string, value: boolean) => void,
-  getLayoutState: (stateKey: string, defaultValue: boolean) => boolean,
-};
+  providerOverrides?: LayoutState
+}
 
-const SearchPageLayoutProvider = ({ getLayoutState, setLayoutState, children }: Props) => {
-  const searchPageLayoutContextValue = useMemo(() => {
-    const config = {
-      sidebar: { isPinned: getLayoutState('sidebarIsPinned', false) },
-    };
-    const actions = { toggleSidebarPinning: () => setLayoutState('sidebarIsPinned', !config.sidebar.isPinned) };
+const defaultState = {
+  sidebar: { isShown: true },
+  viewActionsLayoutOptions: FULL_MENU,
+} as LayoutState;
 
-    return ({
-      config,
-      actions,
-    });
-  }, [getLayoutState, setLayoutState]);
+function SearchPageLayoutProvider({ children, providerOverrides = defaultState }: SearchPageLayoutProviderProps) {
+  const [state] = React.useState<LayoutState>(providerOverrides);
+
+  const value = React.useMemo(() => ({ ...state }), [state]);
```

Comment:
Is this to copy the state before providing it through the context? This is only a shallow copy, so if a consumer manipulates e.g. `state.viewActionsLayoutOptions`, it would still change the original object.

Instead of this, I would suggest solving this through typing the context value as `readonly`.